py/server/deephaven/table.py

```diff
@@ -1656,3 +1664,849 @@ def partitioned_transform(self, other: PartitionedTable, func: Callable[[Table,
             return PartitionedTable(j_partitioned_table=j_pt)
         except Exception as e:
             raise DHError(e, "failed to transform the PartitionedTable with another PartitionedTable.") from e
+
+    def proxy(self, require_matching_keys: bool = True, sanity_check_joins: bool = True) -> PartitionedTableProxy:
+        """Makes a proxy that allows table operations to be applied to the constituent tables of this
+        PartitionedTable.
+
+        Args:
+            require_matching_keys (bool): whether to ensure that both partitioned tables have all the same keys
+                present when an operation uses this PartitionedTable and another PartitionedTable as inputs for a
+                :meth:`~PartitionedTable.partitioned_transform`, default is True
+            sanity_check_joins (bool): whether to check that join operations will only find a given join key in
+                one constituent table and the right table argument if it is also a proxy, default is True
+        """
+        return PartitionedTableProxy(
+            j_pt_proxy=self.j_partitioned_table.proxy(require_matching_keys, sanity_check_joins))
+
+
+class PartitionedTableProxy(JObjectWrapper):
+    """A PartitionedTableProxy is a table operation proxy for the underlying partitioned table. It provides methods
+    that apply table operations on the constituent tables of the underlying partitioned table and produce a new
+    partitioned table from the result constituent tables.
```

Comment:
Does this produce a new partitioned table or partitioned table proxy?

result -> resulting?